control-plane-pki-operator/certificaterevocationcontroller/certificaterevocationcontroller.go

```diff
@@ -0,0 +1,683 @@
+// counts for this controller to be very small (maybe O(10)) and the rate of change to be
+// low, so the extra memory cost in indices is not valuable.
+func signerClassForSecret(secret *corev1.Secret) (certificates.SignerClass, bool) {
+	switch secret.Name {
+	case manifests.CustomerSystemAdminSigner(secret.Namespace).Name:
+		return certificates.CustomerBreakGlassSigner, true
+	default:
+		return "", false
+	}
+}
+
+func secretForSignerClass(namespace string, signer certificates.SignerClass) (*corev1.Secret, bool) {
+	switch signer {
+	case certificates.CustomerBreakGlassSigner:
+		return manifests.CustomerSystemAdminSigner(namespace), true
+	default:
+		return nil, false
+	}
+}
+
+func clientCertSecretForSignerClass(namespace string, signer certificates.SignerClass) (*corev1.Secret, bool) {
+	switch signer {
+	case certificates.CustomerBreakGlassSigner:
+		return manifests.CustomerSystemAdminClientCertSecret(namespace), true
+	default:
+		return nil, false
+	}
+}
+
+func enqueueConfigMap(listCRRs func(namespace string) ([]*hypershiftv1beta1.CertificateRevocationRequest, error)) func(obj runtime.Object) []string {
+	return func(obj runtime.Object) []string {
+		configMap, ok := obj.(*corev1.ConfigMap)
+		if !ok {
+			klog.ErrorS(fmt.Errorf("unexpected object of type %T, wanted %T", obj, &corev1.ConfigMap{}), "could not determine queue key")
+			return nil
+		}
+		signer, ok := signerClassForConfigMap(configMap)
+		if !ok {
+			return nil
+		}
+		return enqueueForSigner(configMap.Namespace, signer, listCRRs)
+	}
+}
+
+// signerClassForConfigMap determines the signer classes that the configmap contains data for.
+// We could use this transformation to create an index, but we expect the scale of resource
+// counts for this controller to be very small (maybe O(10)) and the rate of change to be
+// low, so the extra memory cost in indices is not valuable.
+func signerClassForConfigMap(configMap *corev1.ConfigMap) (certificates.SignerClass, bool) {
+	switch configMap.Name {
+	case manifests.CustomerSystemAdminSignerCA(configMap.Namespace).Name:
+		return certificates.CustomerBreakGlassSigner, true
+	default:
+		return "", false
+	}
+}
+
+func configMapForSignerClass(namespace string, signer certificates.SignerClass) (*corev1.ConfigMap, bool) {
+	switch signer {
+	case certificates.CustomerBreakGlassSigner:
+		return manifests.CustomerSystemAdminSignerCA(namespace), true
+	default:
+		return nil, false
+	}
+}
+
+func (c *CertificateRevocationController) syncCertificateRevocationRequest(ctx context.Context, syncContext factory.SyncContext) error {
+	namespace, name, err := cache.SplitMetaNamespaceKey(syncContext.QueueKey())
+	if err != nil {
+		return err
+	}
+
+	action, requeue, err := c.processCertificateRevocationRequest(namespace, name, nil)
+	if err != nil {
+		return err
+	}
+	if requeue {
+		return factory.SyntheticRequeueError
+	}
+	if action != nil {
+		if err := action.validate(); err != nil {
+			panic(err)
```

Comment:
panic looks improper

Reply:
This is programmer error, and is a) entirely irrecoverable and b) should never merge to `main` - what would you do?